src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfigurationValidator.cs

```diff
@@ -0,0 +1,107 @@
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.IdentityModel.Protocols.OpenIdConnect.Configuration
+{
+    /// <summary>
+    /// Defines a class for validating the OpenIdConnectConfiguration by using default policy.
+    /// </summary>
+    public class OpenIdConnectConfigurationValidator : IConfigurationValidator<OpenIdConnectConfiguration>
+    {
+        private int _minimumNumberOfKeys = DefaultMinimumNumberOfKeys;
+
+        /// <summary>
+        /// 1 is the default minimum number of keys.
+        /// </summary>
+        public static readonly int DefaultMinimumNumberOfKeys = 1;
+
+        /// <summary>
+        /// Validates a OpenIdConnectConfiguration by using current policy.
+        /// </summary>
+        /// <param name="openIdConnectConfiguration">The OpenIdConnectConfiguration to validate.</param>
+        /// <returns>A <see cref="ConfigurationValidationResult"/> that contains validation result.</returns>
+        public ConfigurationValidationResult Validate(OpenIdConnectConfiguration openIdConnectConfiguration)
+        {
+            if (openIdConnectConfiguration == null)
+                throw new ArgumentNullException(nameof(openIdConnectConfiguration));
+
+            if (openIdConnectConfiguration.JsonWebKeySet == null || openIdConnectConfiguration.JsonWebKeySet.Keys.Count == 0)
+            {
+                return new ConfigurationValidationResult
+                {
+                    Exception = new ConfigurationValidationException(LogMessages.IDX21817),
+                    Succeeded = false
+                };
+            }
+
+            if (openIdConnectConfiguration.SigningKeys.Count < MinimumNumberOfKeys)
+            {
+                if (openIdConnectConfiguration.JsonWebKeySet.ConvertKeyInfos.Any())
+                {
+                    return new ConfigurationValidationResult
+                    {
+                        Exception = new ConfigurationValidationException(LogHelper.FormatInvariant(LogMessages.IDX21818, LogHelper.MarkAsNonPII(MinimumNumberOfKeys), string.Join("\n", openIdConnectConfiguration.JsonWebKeySet.ConvertKeyInfos.Select(x => x.Key.Kid.ToString() + ": " + string.Join("; ", x.Value))))),
+                        Succeeded = false
+                    };
+                }
+                else
+                {
+                    return new ConfigurationValidationResult
+                    {
+                        Exception = new ConfigurationValidationException(LogHelper.FormatInvariant(LogMessages.IDX21819, LogHelper.MarkAsNonPII(MinimumNumberOfKeys), LogHelper.MarkAsNonPII(openIdConnectConfiguration.SigningKeys.Count))),
+                        Succeeded = false
+                    };
+                }
+            }
```

Comment:
I think this can be a bit more terse and still pretty clear:

return new ConfigurationValidationResult
                    {
                        Exception = new ConfigurationValidationException(
                            openIdConnectConfiguration.JsonWebKeySet.ConvertKeyInfos.Any() ? 
                            LogHelper.FormatInvariant(LogMessages.IDX21818, LogHelper.MarkAsNonPII(MinimumNumberOfKeys), string.Join("\n", openIdConnectConfiguration.JsonWebKeySet.ConvertKeyInfos.Select(x => x.Key.Kid.ToString() + ": " + string.Join("; ", x.Value)))) :
                            LogHelper.FormatInvariant(LogMessages.IDX21819, LogHelper.MarkAsNonPII(MinimumNumberOfKeys), LogHelper.MarkAsNonPII(openIdConnectConfiguration.SigningKeys.Count))),
                        Succeeded = false
                    };